src/useBaseFetch.js

```diff
@@ -6,12 +6,28 @@ export default config => {
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState('');
   const isMounted = useRef(true);
+
   const { current } = isMounted;
+  const { CancelToken } = axios;
+  const source = CancelToken.source();
+
+  const configWithCancelToken = rawConfig => {
```

Comment:
Could potentially be this instead:

```js
const configWithCancelToken = rawConfig => {
  if (typeof rawConfig === 'string') {
    return {
      url: rawConfig,
      cancelToken: source.token,
    };
  }

  if (rawConfig.constructor === Object) {
    return {
      ...rawConfig,
      cancelToken: source.token,
    };
  }

  return rawConfig;
};
```

Guess we will have to enforce and check what the user passes in